ambari-web/app/views/application.js

```diff
@@ -48,13 +48,25 @@ App.ApplicationView = Em.View.extend({
    */
   initNavigationBar: function () {
     if (App.get('router.mainController.isClusterDataLoaded')) {
-      $('body').on('DOMNodeInserted', '.navigation-bar', () => {
-        $('.navigation-bar').navigationBar({
-          fitHeight: true,
-          collapseNavBarClass: 'icon-double-angle-left',
-          expandNavBarClass: 'icon-double-angle-right'
-        });
-        $('body').off('DOMNodeInserted', '.navigation-bar');
+      const observer = new MutationObserver(mutations => {
+        if (document.querySelector('.navigation-bar')) {
+          observer.disconnect();
+          $('.navigation-bar').navigationBar({
+            fitHeight: true,
+            collapseNavBarClass: 'icon-double-angle-left',
+            expandNavBarClass: 'icon-double-angle-right'
+          });
+        }
```

Comment:
The number of executions can be reduced.
```suggestion
        var targetNode
        if (mutation.type === 'childList' && (targetNode = $('.navigation-bar')).length) {
          observer.disconnect();
          targetNode.navigationBar({
            fitHeight: true,
            collapseNavBarClass: 'icon-double-angle-left',
            expandNavBarClass: 'icon-double-angle-right'
          });
        }
```